src/Runtime/SonataPagePathRuntime.php

```diff
@@ -27,6 +28,21 @@ public function getRunner(?object $application): RunnerInterface
             return parent::getRunner($application);
         }
 
+        Request::setFactory(
```

Comment:
it is the line: https://github.com/sonata-project/SonataPageBundle/blob/4.x/src/Request/RequestFactory.php#L80

Reply:
@VincentLanglet @GeraudBourdin

Reply:
> ok here we go, it was the reason!
> 
> into the RequestFactory, it was calling this code.
> 
> I don't know why they were adding this. does it create a main request? or does it create a new request? I am not sure. But the thing is: it solves the issue.

yes the initial goal was to override the main request.

Reply:
do you have any idea why do we need to do this?

Well it is the same behaviour of RequestFactory, my only point here, is we don't know why do we need this.
why do we need to override the main request 😢 ?

Reply:
do we override to have a main request? well I don't know :(

do you have any opinion here @VincentLanglet ?

Reply:
no override is not the good term. Replace the main request creation with the sonata extended one.

Reply:
by the way issue is fixed and no extra params is needed. So this should be accepted to be merged

Reply:
for your information, it was an implementation from Symfony 2 🤯 

https://github.com/sonata-project/SonataPageBundle/commit/2b70f8156afe64503710e0893c46ed92e8d7d6aa#diff-2d4fd4a3aac457940ec248d6e8421c65f88a2fc4035a3494416b29826bac53e5

Reply:
> by the way issue is fixed and no extra params is needed. So this should be accepted to be merged

yeah I know, I just want to know if we should add this or check if it is a main request on event.
because after this be released, I guess we can't just remove it, probably we need to keep some BC :(

Reply:
I prefer the mainRequest check, but if subrequest is used and if it is, may be siteRequest is needed.  Could this  Request::setFactory thing coould be deprecated for the 5.x release maybe ?

Reply:
> I prefer the mainRequest 

me too, the current approach sounds like workaround, check the main request is more common to see in symfony's doc